pkg/api/internal/rest/request.go

```diff
@@ -359,6 +360,12 @@ func (r *Request) checkResponse(resp *http.Response) *Result {
 		err := json.Unmarshal(body, &jsonErr)
 		if err == nil {
 			err = errors.New(jsonErr.Error)
```

Comment:
the API call is failed, so we return the standard HTTPError unmarshaled from the body if unmarshal is OK
otherwise, we return the HTTP body to use directly if cdc server does not return an HTTPError

Reply:
Go it.